### PR TITLE
Improve Dropdown Menu

### PR DESCRIPTION
**Pull request by YeungKC**
Because even if Scrollbar.isAlwaysShown is true, if it cannot be scrolled, then the scrollbar will not appear and it should not be calculated whether to set Scrollbar.isAlwaysShown to true, so this pr is an improvement to #73654.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by shihaohong** [comment]
@YeungKC I'm actually a little confused too. Could you describe the improvement? I tried reading the test but it just seems like the first time the menu is opened, the scrollbar doesn't show, but the second time, it does?

**Comment by shihaohong** [comment]
Ahh, sorry, I missed the additional zero in the test. Give me a second, going to review the test to improve it a little

File: packages/flutter/test/material/dropdown_test.dart [hunk 1]
```diff
@@ -2921,6 +2921,20 @@ void main() {
   });
 
   testWidgets('Dropdown menu should persistently show a scrollbar if it is scrollable', (WidgetTester tester) async {
+    await tester.pumpWidget(buildFrame(
+      value: '0',
+      items: List<String>.generate(/*length=*/10, (int index) => index.toString()),
```
Comment (shihaohong) [inline packages/flutter/test/material/dropdown_test.dart:2926]:
```suggestion
      // menu is short enough to fit onto the screen.
      items: List<String>.generate(/*length=*/10, (int index) => index.toString()),
```

File: packages/flutter/test/material/dropdown_test.dart [hunk 2]
```diff
@@ -2929,7 +2943,7 @@ void main() {
     await tester.tap(find.text('0'));
     await tester.pumpAndSettle();
 
-    final ScrollController scrollController = PrimaryScrollController.of(tester.element(find.byType(ListView)))!;
+    scrollController = PrimaryScrollController.of(tester.element(find.byType(ListView)))!;
```
Comment (shihaohong) [inline packages/flutter/test/material/dropdown_test.dart:2946]:
```suggestion
    scrollController = PrimaryScrollController.of(tester.element(find.byType(ListView)))!;
    // The scrollbar is shown when the list is longer than the height of the screen.
```

File: packages/flutter/test/material/dropdown_test.dart [hunk 3]
```diff
@@ -2921,6 +2921,20 @@ void main() {
...
+
+    ScrollController scrollController = PrimaryScrollController.of(tester.element(find.byType(ListView)))!;
+    expect(scrollController.position.maxScrollExtent, 0);
+    expect(find.byType(Scrollbar), isNot(paints..rect()));
+
+    await tester.tap(find.text('0').last);
+    await tester.pumpAndSettle();
     await tester.pumpWidget(buildFrame(
       value: '0',
       items: List<String>.generate(/*length=*/100, (int index) => index.toString()),
```
Comment (shihaohong) [inline packages/flutter/test/material/dropdown_test.dart:2940]:
```suggestion
      // menu is too long to fit onto the screen.
      items: List<String>.generate(/*length=*/100, (int index) => index.toString()),
```

**Review by shihaohong — APPROVED** [review]
after nits addressed, LGTM!

**Comment by YeungKC** [comment]
Because even if Scrollbar.isAlwaysShown is true, if it is not scrollable, then the scrollbar will not appear, and I shouldn't have to calculate whether to set Scrollbar.isAlwaysShown to true myself.

**Comment by shihaohong** [comment]
The plugins check failing is unrelated to the PR. I've already filed an issue about it and hopefully the team fixes it in the next day or so

**Comment by shihaohong** [comment]
You'll need to rebase with the `master` branch, since it contains the commit that will pass the `Linux flutter_plugins` check. Sorry for the trouble!

**Pull request merged**